src/org/labkey/test/components/ui/files/FileUploadPanel.java

```diff
@@ -18,11 +18,13 @@ public class FileUploadPanel extends WebDriverComponent<FileUploadPanel.ElementC
 {
     final WebElement _el;
     final WebDriver _driver;
+    final boolean _backwardsCompat;
 
-    public FileUploadPanel(WebElement element, WebDriver driver)
+    public FileUploadPanel(WebElement element, WebDriver driver, boolean backwardsCompat)
```

Comment:
I don't love the pattern of parameterizing backwards compatibility here (it seems to require a bit of extra complexity) but I think it does have the upside of making it simple in the future to know when no more callers needing back-support exist.

The other alternative might be to use an xpath-union locator that supports both cases, like 
`Locator.XPathLocator.union(
    Locator.tagWithClass("div", "attached-file--container")
                .withChild(Locator.tagWithClass("span", "fa-times-circle")), 
    Locator.tagWithClass("div", "attached-file__container"))` 
and skip the conditional locator getter